packages-common/pluggable-widgets-generator/generators/app/index.js

```diff
@@ -0,0 +1,284 @@
+        this._copyFile(`commons/eslintrc.${this.widget.isLanguageTS ? "ts" : "js"}.js`, ".eslintrc.js");
+        this._copyFile("commons/prettier.config.js", "prettier.config.js");
+        this._copyFile("commons/.gitattributes", ".gitattributes");
+
+        if (this.widget.license) {
+            switch (this.widget.license.toLowerCase()) {
+                case "apache-2.0":
+                    this._copyTemplate("licenses/APACHE2", "LICENSE");
+                    break;
+                case "mit":
+                    this._copyTemplate("licenses/MIT", "LICENSE");
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+    _writeUnitTests() {
+        const extension = this.widget.isLanguageTS ? "tsx" : "jsx";
+
+        if (this.widget.hasUnitTests) {
+            if (this.widget.isPlatformWeb) {
+                if (this.widget.usesFullTemplate) {
+                    this._copyFile(
+                        `${this.widget.templateSourcePath}${widgetSrcFolder}/__tests__/Alert.spec.${extension}.ejs`,
+                        `${widgetSrcFolder}/__tests__/Alert.spec.${extension}`
+                    );
+                    this._copyTemplate(
+                        `${this.widget.templateSourcePath}${widgetSrcFolder}/__tests__/BadgeSample.spec.${extension}.ejs`,
+                        `${widgetSrcFolder}/__tests__/BadgeSample.spec.${extension}`
+                    );
+                } else {
+                    this._copyFile(
+                        `${this.widget.templateSourcePath}${widgetSrcFolder}/__tests__/HelloWorldSample.spec.${extension}.ejs`,
+                        `${widgetSrcFolder}/__tests__/HelloWorldSample.spec.${extension}`
+                    );
+                }
+            } else {
+                if (this.widget.usesFullTemplate) {
+                    this._copyFile(
+                        `${this.widget.templateSourcePath}${widgetSrcFolder}/__tests__/BadgeSample.spec.${extension}.ejs`,
+                        `${widgetSrcFolder}/__tests__/BadgeSample.spec.${extension}`
+                    );
+                } else {
+                    this._copyFile(
+                        `${this.widget.templateSourcePath}${widgetSrcFolder}/__tests__/HelloWorldSample.spec.${extension}.ejs`,
+                        `${widgetSrcFolder}/__tests__/HelloWorldSample.spec.${extension}`
+                    );
+                }
+            }
+        }
+    }
+
+    _writeEndToEndTests() {
+        const widgetName = this.widget.name;
+        const extension = this.widget.isLanguageTS ? "ts" : "js";
+
+        if (this.widget.hasE2eTests && this.widget.isPlatformWeb) {
+            if (this.widget.isLanguageTS) {
+                this._copyFile("typings/WebdriverIO.d.ts", "tests/e2e/typings/WebdriverIO.d.ts");
+                this._copyFile(this.widget.templateSourcePath + "tests/e2e/tsconfig.json", "tests/e2e/tsconfig.json");
```

Comment:
use template string here as in other paths?

Reply:
Refactored